### PR TITLE
[decoupled-execution] Integration Part 4 - More tests

### PR DESCRIPTION
**Pull request by yuxiamit**
Add unit tests for ordering_state_computer and execution_phase.

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update or suggest changes to the docs at https://developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: execution/executor-types/src/lib.rs [hunk 1]
```diff
@@ -153,6 +155,25 @@ impl StateComputeResult {
...
+            parent_num_leaves: 0,
+            epoch_state: None,
+            compute_status: vec![],
+            transaction_info_hashes: vec![],
+            reconfig_events: vec![],
+            signature: None,
+        }
+    }
+
+    pub fn new_dummy() -> Self {
```
Comment (zekun000) [inline execution/executor-types/src/lib.rs:174]: add some comments how this is used in ordering state computer

File: consensus/src/test_utils/mock_state_computer.rs [hunk 2]
```diff
@@ -106,13 +96,52 @@ pub struct EmptyStateComputer;
...
+#[async_trait::async_trait]
+impl StateComputer for RandomComputeResultStateComputer {
     fn compute(
         &self,
         _block: &Block,
         _parent_block_id: HashValue,
     ) -> Result<StateComputeResult, Error> {
         Ok(StateComputeResult::new(
-            *ACCUMULATOR_PLACEHOLDER_HASH,
+            self.random_compute_result_root_hash,
```
Comment (zekun000) [inline consensus/src/test_utils/mock_state_computer.rs:144]: new_dummy_with_root_hash?

File: execution/executor-types/src/lib.rs (hunk 1 above)
Comment (yuxiamit) [reply execution/executor-types/src/lib.rs:174]: resolved

File: consensus/src/test_utils/mock_state_computer.rs (hunk 2 above)
Comment (yuxiamit) [reply consensus/src/test_utils/mock_state_computer.rs:144]: resolved

**Review by zekun000 — APPROVED** [review]

**Pull request merged**